test/inductor/test_mkldnn_pattern_matcher.py

```diff
@@ -3023,6 +3025,93 @@ def matcher_check_fn():
                 rtol=0.07,
             )
 
+    @skipIfNoDynamoSupport
+    @skipIfNoONEDNN
+    # TODO: investigate options of torch.compile in fbcode
+    @unittest.skipIf(IS_FBCODE, "Failing in fbcode")
+    @parametrize("has_bias", [True, False])
+    @parametrize("dtype", [torch.float, torch.bfloat16])
+    @parametrize("per_channel_quant", [True, False])
+    @parametrize("dynamic", [True, False])
+    def test_smooth_quant_with_int_mm(
+        self, has_bias, dtype, per_channel_quant, dynamic
+    ):
+        r"""
+        This testcase check if we can match the SmoothQuant int8 linear pattern from Torchao.
+        The pattern is:
+            (no bias) reshape -> _int_mm -> convert_element_type -> (expand -> mul) -> mul -> reshape
```

Comment:
@leslie-fang-intel, same comment here as https://github.com/pytorch/pytorch/pull/141851#discussion_r1866946236?

Reply:
@sanchitintel What pattern did you see?

Reply:
@leslie-fang-intel's concern seems to be that this visual depiction indicates that the output of `torch._int_mm` is fed into `expand`

Reply:
@Xia-Weiwen, I now think that @leslie-fang-intel was asking me to include `expand -> mul` within brackets, just like you have.

@leslie-fang-intel, also, why not use indicate here that `expand` is optional?

Reply:
Ok